src/chocolatey/infrastructure.app/services/NugetService.cs

```diff
@@ -845,13 +845,104 @@ public void remove_rollback_directory_if_exists(string packageName)
             return packageInstalls;
         }
 
-        /// <summary>
-        /// Sets the configuration for the package upgrade
-        /// </summary>
-        /// <param name="config">The configuration.</param>
-        /// <param name="packageInfo">The package information.</param>
-        /// <returns>The original unmodified configuration, so it can be reset after upgrade</returns>
-        private ChocolateyConfiguration set_package_config_for_upgrade(ChocolateyConfiguration config, ChocolateyPackageInformation packageInfo)
+		public ConcurrentDictionary<string, PackageResult> get_outdated(ChocolateyConfiguration config)
+		{
+			var packageManager = NugetCommon.GetPackageManager(
+				config,
+				_nugetLogger,
+				_packageDownloader,
+				installSuccessAction: null,
+				uninstallSuccessAction: null,
+				addUninstallHandler: false);
+
+			var outdatedPackages = new ConcurrentDictionary<string, PackageResult>();
+
+			set_package_names_if_all_is_specified(config, () => { config.IgnoreDependencies = true; });
+			var packageNames = config.PackageNames.Split(new[] { ApplicationParameters.PackageNamesSeparator }, StringSplitOptions.RemoveEmptyEntries).or_empty_list_if_null().ToList();
+
+			foreach (var packageName in packageNames)
+			{
+				var repository = packageManager.SourceRepository;
+
+				bool isRemote = IsRepositoryRemote(repository);

```

Comment:
This call and the repository seem like they are not a per iteration kind of call. Might be best not to recheck that each time.